tests/acceptance/__init__.py

```diff
@@ -54,108 +74,127 @@ def get_random_suffix():
+                "visibility": "internal",
+            }
+        )
+    if TYPE_CHECKING:
+        assert isinstance(group, Group)
+    return group
 
 
 def create_groups(group_base_name, no_of_groups):
     groups = []
     for group_no in range(1, no_of_groups + 1):
         group_name = group_base_name + str(group_no)
         try:
-            gl.get_group_case_insensitive(group_name)
-        except NotFoundException:
-            gl.create_group(group_name, group_name)
-        groups.append(group_name)
+            group = gl.groups.get(group_name)
+        except gitlab.GitlabGetError:
+            group = create_group(group_name)
+        groups.append(group)
     return groups
 
 
 def delete_groups(group_base_name, no_of_groups):
     for group_no in range(1, no_of_groups + 1):
         group_name = group_base_name + str(group_no)
-        gl.delete_group(group_name)
-
-
-def create_project(group_name, project_name):
-    group = gl.get_group(group_name)
-    gl.create_project(
-        project_name,
-        project_name,
-        group["id"],
-        default_branch="main",
-        wait_if_still_being_deleted=True,
+        with allowed_codes(404):
+            gl.groups.delete(group_name)
+
+
+def create_project(group: Group, project_name) -> Project:
+    project = gl.projects.create(
+        {
+            "name": project_name,
+            "path": project_name,
+            "namespace_id": group.id,
+            "default_branch": "main",
+        }
     )
 
-    gl.add_file(
-        f"{group_name}/{project_name}",
-        "main",
-        "README.md",
-        DEFAULT_README,
-        "Create README",
+    project.files.create(
+        {
+            "branch": "main",
+            "file_path": "README.md",
+            "content": DEFAULT_README,
+            "commit_message": "Create README",
+        }
     )
+    if TYPE_CHECKING:
+        assert isinstance(project, Project)
+    return project
 
 
 def create_users(user_base_name, no_of_users):
     users = []
     for user_no in range(1, no_of_users + 1):
         username = user_base_name + str(user_no)
+        existing = gl.users.list(username=username)
         try:
-            gl.get_user_by_name(username)
-        except NotFoundException:
-            gl.create_user(
-                username + "@example.com", username + " Example", username, "password"
+            user = existing[0]
+        except IndexError:
+            user = gl.users.create(
+                {
+                    "username": username,
+                    "email": username + "@example.com",
+                    "name": username + " Example",
+                    "password": username + "password",
+                }
             )
-        users.append(username)
+        users.append(user)
     return users
 
 
-def delete_users(user_base_name, no_of_users):
-    for user_no in range(1, no_of_users + 1):
-        username = user_base_name + str(user_no)
-        gl.delete_user(username)
```

Comment:
Removed as we can delete the created user objects directly via `user.delete()` instead of searching for them.

Similar for most of the deleted helpers below.